### DISCUSSION TITLE
HyperHDR + HyperSerialESP32 = no LEDs

### DISCUSSION THREAD

**Discussion by mraneri** [awawa-dev/HyperHDR#378] [General]
My setup is:
- A strip of SK6812 RGBNW, 274 total leds.
- HyperHDR on a pi Zero W with a USB hub that has my grabber and ESP32 (9102x) connected to it.
The pi and LEDs are powered by the same power supply with the data wire plugged into the D2/GPIO2 pin. 
The ESP32 is flashed with v6.1.0.0 HyperSerialESP32_SK6812_NEUTRAL_WHITE_FIRSTLED_DISABLED.bin (ENABLED behaves the same way)
[HyperHDR settings](https://i.imgur.com/lQPhRkc.png) pictured.
The LEDs don't do anything.

Now, if I use RPi PWM and run the data off of the pi or if I flash the ESP32 with HyperSerialWLED everything works just fine. I have tried a couple different USB hubs and several cables. Any Ideas?

**Comment by awawa-dev** [comment]
Hi
I doubt it's related to USB because HyperSerialESP32 is using much less power than HyperSerialWLED since the wifi module is disabled. Run the diagnostic test for HyperSerialESP32.
1 run some effects in HyperHDR using HyperSerialESP32
2 close HyperHDR
3 connect using serial port client to HyperSerialESP32 (speed 2000000)
4 read the statistics

**Reply by mraneri** [reply]
Effects do not work either.

This is what PuTTY shows:

▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒#▒$$▒▒▒<▒▒'<▒▒▒'<<$▒▒
#$'▒?'▒▒$▒;▒▒▒▒▒$▒''▒<▒▒'▒▒#$▒'<▒▒▒▒#▒▒;#$ ▒;▒▒▒$▒▒▒$▒$▒▒▒8▒▒$;▒ܜ8▒▒▒8▒$ #
▒;▒▒▒8▒▒$;▒l▒▒▒▒▒▒$▒▒
;▒▒▒8▒$▒$ ▒
▒l▒▒l▒▒;▒ܜ8▒▒▒<▒$ ;▒▒▒8▒▒▒d;▒▒▒8▒▒▒$▒▒
$▒▒$ ▒▒▒$▒▒▒$#▒▒▒#
$$▒▒ ▒$$  ▒▒#$`▒▒$▒▒▒$▒<▒▒▒$$▒▒$ ?▒▒▒▒▒▒▒▒▒▒$ ;$$▒▒▒▒8▒▒▒$▒▒$ ;$;'▒<;▒#▒▒;▒▒▒8▒▒▒8▒▒▒▒8▒$▒;▒▒▒8▒▒▒▒$ ;$▒▒#▒''▒▒<▒▒###$
Welcome!
Awa driver 6.
First LED: disabled
Default color mode: RGBW neutral
RGBW calibration. White limit(%): 100.00 %, red: 176 , green: 176 , blue: 112
HyperSerialESP32 version 6.1
Statistics for the last full 1 second cycle.
Frames per second: 0
Good frames: 0
Bad frames:  0

**Reply by awawa-dev** [reply]
Maybe I didn't specify it: you can't reset or disconnect ESP32 during this test because the statistics are lost then. And there is a trace of esp32 restart in the logs. Similarly, when you run HyperHDR, the serial port client cannot be running at the same time (you will get a critical error in HyperHDR because the port will be busy). Do you use the binary from the HyperSerialESP32 website or do you compile it yourself?

**Reply by mraneri** [reply]
Well, it's working now. I had to install screen on my pi, rebooted, and it started working. That's at least the tenth time I've restarted the pi today.
It does seem to be persistent after reboots now too.

**Reply by mraneri** [reply]
Thank you for the super quick response!
